src/ray/raylet/scheduling/local_resource_manager.cc

```diff
@@ -429,9 +429,16 @@ std::optional<syncer::RaySyncMessage> LocalResourceManager::CreateSyncMessage(
     resources_data.set_object_pulls_queued(resources.object_pulls_queued);
   }
 
-  const auto now = absl::Now();
-  resources_data.set_idle_duration_ms(
-      absl::ToInt64Milliseconds(now - GetResourceIdleTime().value_or(now)));
+  auto idle_time = GetResourceIdleTime();
+  if (idle_time.has_value()) {
+    // We round up the idle duration to the nearest millisecond such that the idle
+    // reporting would be correct even if it's less than 1 millisecond.
+    // This is needed since we will not be reporting resource usage if there's no more
+    // change due to lightweight resource reporting.
```

Comment:
```suggestion
    // We round UP the idle duration to the nearest millisecond such that the idle
    // reporting would be correct even if it's less than 1 millisecond.
    // Otherwise, we will report idle millisecond as 0 and autoscaler will think that the node is not idle which is wrong.```

I feel we don't need to mention lightweight resource reporting: regardless of whether we have lightweight or heavyweight resource reporting, we need to make sure every idle report is correct.